tests/unit/test_charm.py

```diff
@@ -138,6 +144,23 @@ def test_default_scrape_config_is_always_set(self):
         prometheus_scrape_config = scrape_config(config, "prometheus")
         self.assertIsNotNone(prometheus_scrape_config, "No default config found")
 
+    def test_honor_labels_is_always_set_in_scrape_configs(self):
```

Comment:
Missing test cases:
- honor_labels is preset to False
- jobs spec is empty

Reply:
> honor_labels is preset to False

Quite literally cannot happen.

Reply:
Prometheus charm (consumer side) technically has no control over this, so it very much _can_ happen.

Reply:
Because jobs even on the provider side are sanitized using `ALLOWED_KEYS` ( kudos @mmanciop ) the only way the relation data can have `honor[labels]=False` is if relation data is manipulated directly by the provider charm rather than using the library. If there are the types of cases we would like to protect against then I claim I have already won our bug bounty competition for all the charms we ever wrote. :-)